packages/medusa/src/services/totals.ts

```diff
@@ -249,19 +363,22 @@ class TotalsService extends BaseService {
     return discounts
   }
 
-  getLineDiscounts(cart, discount) {
+  getLineDiscounts(
+    cart: Cart | Order,
+    discount: Discount
+  ): LineDiscountAmount[] {
     const subtotal = this.getSubtotal(cart, { excludeNonDiscounts: true })
 
-    let merged = [...cart.items]
+    let merged: LineItem[] = [...cart.items]
 
     // merge items from order with items from order swaps
-    if (cart.swaps && cart.swaps.length) {
+    if ("swaps" in cart && cart.swaps.length) {
```

Comment:
These will likely always be true because typescript, use something link `typeof cart.swaps !== 'undefined'` instead 😉

Reply:
The type of cart here is actually `Cart | Order` and it is only the `Order` type that has a swap property so this one essentially checks whether we are dealing with a `Cart` or `Order` and not as much if the value is defined; although that might be a good check to add too actually :D